src/shared/components/dropdown/BasicDropdown.tsx

```diff
@@ -42,59 +49,158 @@ const BasicDropdown: React.FC<BasicDropdownProps> = ({
+      }
+    }
+  };
+
+  const onTextInputChange = (_event: React.FormEvent<HTMLInputElement>, value: string) => {
+    setFilterValue(value);
+
+    if (value !== selected) {
+      onChange('');
+    }
+  };
+
+  const typeaheadMenu = (
+    <TextInputGroup isPlain>
+      <TextInputGroupMain
+        value={selected ? selected : filterValue}
+        onClick={onToggleClick}
+        onChange={onTextInputChange}
+        id="basic-dropdown-typeahead-input"
+        autoComplete="off"
+        innerRef={textInputRef}
+        placeholder={placeholder}
+        role="combobox"
+        isExpanded={isOpen}
+        aria-controls="basic-dropdown-listbox"
+      />
+      <TextInputGroupUtilities
+        {...(!(selected || filterValue) ? { style: { display: 'none' } } : {})}
+      >
+        <Button variant="plain" onClick={onClearButtonClick} aria-label="Clear input value">
+          <TimesIcon aria-hidden />
+        </Button>
+      </TextInputGroupUtilities>
+    </TextInputGroup>
+  );
+
+  const defaultMenu = selected ? (
+    <>
+      {selected}
+      {selected === recommended && (
+        <>
+          &nbsp;<Badge isRead>Recommended</Badge>
+        </>
+      )}
+    </>
+  ) : (
+    <span>{placeholder ?? ''}</span>
+  );
+
+  const menuToggle = (toggleRef) => {
+    return (
+      <MenuToggle
+        aria-disabled={disabled}
+        data-test="dropdown-toggle"
+        style={{ width: fullWidth ? '100%' : undefined }}
+        ref={toggleRef}
+        isExpanded={isOpen}
+        isDisabled={disabled}
+        onClick={() => onToggleClick()}
+        variant={variant}
+        aria-invalid={validated === ValidatedOptions.error}
+      >
+        {variant === 'default' ? defaultMenu : typeaheadMenu}
+      </MenuToggle>
+    );
+  };
+
   return (
-    <div>
+    <>
       <Select
+        id="basic-dropdown"
         isOpen={isOpen}
         onOpenChange={(newIsOpen) => {
           if (!disabled) setIsOpen(newIsOpen);
         }}
         selected={selected}
         onSelect={handleSelect}
         aria-invalid={validated === ValidatedOptions.error}
-        toggle={(toggleRef) => (
-          <MenuToggle
-            aria-disabled={disabled}
-            data-test="dropdown-toggle"
-            style={{ width: fullWidth ? '100%' : undefined }}
-            ref={toggleRef}
-            isExpanded={isOpen}
-            isDisabled={disabled}
-            onClick={() => !disabled && setIsOpen(!isOpen)}
-            variant="default"
-            aria-invalid={validated === ValidatedOptions.error}
-          >
-            {selected ? (
-              <>
-                {selected}
-                {selected === recommended && (
-                  <>
-                    &nbsp;<Badge isRead>Recommended</Badge>
-                  </>
-                )}
-              </>
-            ) : (
-              <span>{placeholder ?? ''}</span>
-            )}
-          </MenuToggle>
-        )}
+        maxMenuHeight="35rem"
```

Comment:
i guess hardcode is not good here. IMO, the percentage or some pf format is better.